### PR TITLE
Fix minor bugs in treemap charts

### PR DESCRIPTION
**Pull request by jeffwheeler**
I am fixing two bugs in the treemap viewer with this commit.

The first bug is with tooltip rendering. The existing code uses the `opacity` property to hide text when it is too small to fit, but because the text is still present (just invisible to the user), the tooltip still renders over the entire invisible area of the text. Instead, it should render based on the rectangle region, not the invisible text. The fix for this is simple, change `opacity` to `visibility`.

Screen Shot 2021-05-22 at 3 52 37 PM

In the above image, the tooltip is showing the text for the thin rectangle on the right, which is an error.

The second bug is caused by the use of strokes on each rectangle. The total height of the columns sometimes varies, so that one column will be an extra pixel taller, as seen on the following image:

Screen Shot 2021-05-22 at 3 52 49 PM

To resolve this, I use `paddingInternal` to create padding, instead of a stroke. I guess the variable `--color-treemap-text` was originally meant for a different purpose (`text`), but was now being used for the white stroke color. I've removed it.

### CODE REVIEW

File: frontend/src/charts/Treemap.svelte [hunk 1]
```diff
@@ -57,11 +57,11 @@
     <g
       transform={`translate(${d.x0},${d.y0})`}
       use:followingTooltip={() => tooltipText(d)}
+      on:click={() => router.navigate(urlFor(`account/${d.data.account}/`))}
     >
       <rect fill={fill(d)} width={d.x1 - d.x0} height={d.y1 - d.y0} />
       <text
-        use:setOpacity={d}
-        on:click={() => router.navigate(urlFor(`account/${d.data.account}/`))}
```
Comment (yagebu) [inline frontend/src/charts/Treemap.svelte:64]: I believe this was intentional to not make the whole chart clickable. Can you revert this change, which is unrelated to the bug fixes AFAICT?

**Review by yagebu — CHANGES_REQUESTED** [review]
Thanks for these fixes :)

File: frontend/src/charts/Treemap.svelte (hunk 1 above)
Comment (jeffwheeler) [reply frontend/src/charts/Treemap.svelte:64]: Ah, you're correct that this wasn't described in the original pull request comment.

With the original behavior, it was still possible to click on the center of small rectangles because the text was present but transparent. Changing the text to hidden means that it becomes impossible to click without this part of the change. But the original behavior was buggy, because text labels often overlapped (invisibly).

I think that this is probably the cleanest way to go. The original click targets were basically random near small rectangles.

Comment (yagebu) [reply frontend/src/charts/Treemap.svelte:64]: As I've said, the non-clickability of most of the area was intentional (and hence the use of opacity to make even hidden account names clickable), see https://github.com/beancount/fava/issues/490#issuecomment-294529035 for some context. I'd be fine if accounts whose name isn't shown just aren't clickable anymore.

File: frontend/src/charts/Treemap.svelte [hunk 2]
```diff
@@ -38,14 +38,14 @@
...
+  function setVisibility(
     node: SVGTextElement,
     param: HierarchyRectangularNode<AccountHierarchyDatum>
   ) {
     function update(d: HierarchyRectangularNode<AccountHierarchyDatum>) {
       const length = node.getComputedTextLength();
-      node.style.opacity =
-        d.x1 - d.x0 > length + 4 && d.y1 - d.y0 > 14 ? "1" : "0";
+      node.style.visibility =
+        d.x1 - d.x0 > length + 4 && d.y1 - d.y0 > 14 ? "visibile" : "hidden";
```
Comment (yagebu) [inline frontend/src/charts/Treemap.svelte:48]: Type

File: frontend/src/charts/Treemap.svelte (hunk 1 above)
Comment (jeffwheeler) [reply frontend/src/charts/Treemap.svelte:64]: OK, I've changed per your request.

I understand where you're coming from, but it seems very counter to Fitts's Law to minimize the clickable area. Anyways, you get to have what you want. :)

File: frontend/src/charts/Treemap.svelte (hunk 2 above)
Comment (jeffwheeler) [reply frontend/src/charts/Treemap.svelte]: Fixed, thanks

File: frontend/src/charts/Treemap.svelte (hunk 1 above)
Comment (yagebu) [reply frontend/src/charts/Treemap.svelte:64]: > but it seems very counter to Fitts's Law to minimize the clickable area. 

Sure, but I currently don't really see a simple alternative without making these chart rather unusable on mobile.
